src/components/MenuComponent.js

```diff
@@ -18,13 +19,7 @@ class Menu extends Component {
     renderDish(dish) {
```

Comment:
I think generally things like this are not needed. 

Canonically to conditionally render you'd put something like `{dish && <DishDetail dish={dish} />}` instead of `{renderDish()}`

Also you should return `null` if nothing should be put in the dom. Adding extra elements is generally a bad practice (because of css / layout issues that can arise)

Reply:
+1 on the render functions comment.